### ISSUE TITLE
Only select files filtered by search

### ISSUE THREAD

**Issue by raimund-schluessler** [nextcloud/server#2075]
### Steps to reproduce
1. Have a folder with various files.
![screen shot 2016-11-10 at 13 24 50](https://cloud.githubusercontent.com/assets/2496460/20176676/2b31bfd8-a749-11e6-92b4-df001a86dd01.png)

2. Search within the folder (e.g. search for "mp3")
![screen shot 2016-11-10 at 13 24 59](https://cloud.githubusercontent.com/assets/2496460/20176682/34bd0b52-a749-11e6-9b45-12d0db8930d2.png)

3. Select all files using the "select_all_files" checkbox.
![screen shot 2016-11-10 at 13 25 07](https://cloud.githubusercontent.com/assets/2496460/20176684/3755bd28-a749-11e6-9382-a1d9d6e10a61.png)

### Expected behaviour
I would expect that only the files shown after searching are selected. This would e.g. simplify downloading only certain files.

### Actual behaviour
Instead of only selecting the 3 shown files, all files within the folder (also the ones filtered out by searching) are selected.

**Nextcloud version:** 10.0.1

I searched for similar issues and found this one: https://github.com/nextcloud/server/issues/1274 However, I think this issue here is a bit different as this is not about group actions. Correct me if I'm wrong.

@icewind1991 @jancborchardt

**Comment by jasonbayton** [comment]
I can also add when using Shift+click to select multiple files, the same sort of thing happens; it selects far more than it should presumably as it's selecting files between those filtered.

**Comment by Schmuuu** [comment]
For me this is actually a serious bug. I was just going to open a new issue (but found this one then), because the following happened to me:

With every photo I take on my smartphone there are additional files (*.nar, *.thm, ...) created and all this files together with the photos itself are then automatically uploaded to NC.
When I wanted to clean up my upload folder on the server, I performed a search for the specific file extensions and hit the "select all" checkbox not knowing about this bug. I only knew I have a few thousands photos on my server, but no exact number and when it read "selected 3147 files" I thought, this could be right. 

Well, after hitting delete, I found my Photo folder to be completely empty. Older photos have been deleted in the trash bin as I exceeded the Quota limit. I got a backup elsewhere, but that wasn't fun and rather a big shock at first. 

Is there a chance to have this fixed earlier than NC13? 
While the current behavior is everything but obvious I can only hope it doesn't happen to other users, who don't take backups serious.
